resources/lib/downloadutils.py

```diff
@@ -443,6 +443,9 @@ def downloadUrl(self, url, suppress=False, postBody=None, method="GET", authenti
             filter_string = getDetailsString()
             url = url.replace("{field_filters}", filter_string)
 
+        if url.find(" / ") != -1:
```

Comment:
this should be done only for the values that are effected not the full url.
what values/params are you having issues with?

Reply:
It effects Genres and Studios.

Reply:
where are the URLs coming from? Where in EmbyCon are they being built?

Reply:
The URLs would be build by Kodi skins.
I'm mainly interested in genres, because the default Emby query for similar movies is returning totally crap results. It's the same in the Emby webif, so not a fault of your plugin.

Reply:
ok I see, this is a problem them as I have no control over what people submit to the addon.
I think this is going to have to be done in the submitter, it is a slipry slow, now " / " and then all sorts of stuff as more dirty data is added.
This should really be up to the creator of the URL to make sure it is ok to be used.

Reply:
What about this, does it fullfill your needs:
 ```
import re

if re.search("(?i)Genres=", url):
    genreMatch = re.search(r'(?i)Genres=[\w\s/]+', url)
    if genreMatch:
        genreReplaceString = genreMatch.string.replace(" / ", "|")
        url = url.replace(genreMatch.string, genreReplaceString)
```

Reply:
I would like to know where these links are coming from and why we can t fix it there?
Which skin is producing them?
Where in the skin is it being produced?
Why cant it just use Thriller|Horror instead of Thriller / Horror?

Reply:
It's not in any skin yet, but it will hopefully get into Embuary's "More like this" section.

`plugin://plugin.video.embycon/?mode=GET_CONTENT&media_type=Movies&url={server}/emby/Items/$INFO[ListItem.Property(id)]/Similar?userId={userid}&Limit={ItemLimit}&format=json&ImageTypeLimit=1&IsMissing=False&Filters={field_filters}`

should then be replaced by

`plugin://plugin.video.embycon/?media_type=Movies&mode=GET_CONTENT&url={server}/emby/Users/{userid}/Items?genres=$INFO[ListItem.Genre]&IncludeItemTypes=Movie&Recursive=true&CollapseBoxSetItems=true&IsMissing=false&ImageTypeLimit=1&SortBy=Random&Limit={ItemLimit}&format=json`

$INFO[ListItem.Genre] is a dynamic string label in Kodi. It changes with every focused item.
But if there is more then one genre for an item, then Kodi concatenates them always to "Genre1 / Genre2 / Genre3" etc. And as far as i know there's no string replacement function build into Kodi's skin architecture. So it has to be done either by another Kodi script addon or in the addon where the variable should be used.

Reply:
if these ListItems are from EmbyCon then it is the one responsible for populating them

This is the code it uses to build the genre list for a ListItem
    genres = item["Genres"]
    if genres is not None and len(genres) > 0:
        item_details.genre = " / ".join(genres)

So EmbyCon could just set a new property on the ListItem like this

list_item.setProperty('genres', "|".join(genres))

and you could use it in your Skin like this

$INFO[ListItem.Property(genres)]

Reply:
I've did a quick test, but this would return all Emby genres and not the item specific ones.
I've defined list_item.setProperty('genres', "|".join(genres)) in functions.py
but maybe that is wrong?!

Reply:
no it would need to be set per item in the item_functions.py file.
INstead of storing the Genre string you would store a list of genres and then do the string join on them on the add_gui_item function.

Reply:
I am still confused by this, where would you use this, you would be building a new url based on the genre of an existing displayed item? That does not make much sense.
Where is this list of original ListItems coming from? Is it a list of movies? As are you going to make a skin link that says something like "Show more of the type of movie" or something?

Reply:
Seems it only works with "%7C".join but not with "|".join but now i got it!

Reply:
> I am still confused by this, where would you use this, you would be building a new url based on the genre of an existing displayed item? That does not make much sense.
> Where is this list of original ListItems coming from? Is it a list of movies? As are you going to make a skin link that says something like "Show more of the type of movie" or something?

Yes "More like this" already there in the movie details screen in Embuary. But Emby's "similar movies query" that is currently in use, returns nonsense. That's why a custom genre query would be better.